sockeye/inference.py

```diff
@@ -137,8 +137,8 @@ def initialize(self, max_input_length: int, get_max_output_length_function: Call
         self.decoder_module.init_params(arg_params=self.params, allow_missing=False)
 
         if self.cache_output_layer_w_b:
-            self.output_layer_w = self.params[self.decoder.output_layer.w.name].asnumpy()
-            self.output_layer_b = self.params[self.decoder.output_layer.b.name].asnumpy()
+            self.output_layer_w = mx.nd.array(self.params[self.decoder.output_layer.w.name], ctx=self.context)
```

Comment:
this always copies even if source and target context are the same.
You could use ndarray.as_in_context() instead:
https://mxnet.incubator.apache.org/api/python/ndarray/ndarray.html#mxnet.ndarray.NDArray.as_in_context